src/thermo/PlasmaPhase.cpp

```diff
@@ -268,4 +268,94 @@ void PlasmaPhase::updateThermo() const
     m_g0_RT[k] = m_h0_RT[k] - m_s0_R[k];
 }
 
+double PlasmaPhase::enthalpy_mole() const {
+    double value = IdealGasPhase::enthalpy_mole();
+    value += GasConstant * (electronTemperature() - temperature()) *
+             moleFraction(m_electronSpeciesIndex) *
+             m_h0_RT[m_electronSpeciesIndex];
+    return value;
+}
+
+void PlasmaPhase::getGibbs_ref(double* g) const
+{
+    IdealGasPhase::getGibbs_ref(g);
+    g[m_electronSpeciesIndex] *= electronTemperature() / temperature();
+}
+
+void PlasmaPhase::getStandardVolumes_ref(double* vol) const
+{
+    IdealGasPhase::getStandardVolumes_ref(vol);
+    vol[m_electronSpeciesIndex] *= electronTemperature() / temperature();
+}
+
+void PlasmaPhase::getPartialMolarEnthalpies(double* hbar) const
+{
+    IdealGasPhase::getPartialMolarEnthalpies(hbar);
+    hbar[m_electronSpeciesIndex] *= electronTemperature() / temperature();
+}
+
+void PlasmaPhase::getPartialMolarEntropies(double* sbar) const
+{
+    IdealGasPhase::getPartialMolarEntropies(sbar);
+    double logp = log(pressure() / refPressure());
+    double logpe = log(electronPressure() / refPressure());
+    sbar[m_electronSpeciesIndex] += GasConstant * (logp - logpe);
+}
+
+void PlasmaPhase::getPartialMolarIntEnergies(double* ubar) const
+{
+    const vector_fp& _h = enthalpy_RT_ref();
+    for (size_t k = 0; k < m_kk; k++) {
+        if (k == m_electronSpeciesIndex) {
+            ubar[k] = RTe() * (_h[k] - 1.0);
+        } else {
+            ubar[k] = RT() * (_h[k] - 1.0);
+        }
```

Comment:
As a minor optimization, you can avoid checking all the `if` conditions here and just evaluate the value using the gas temperature for all species, and then replace the value for the electron species after the loop.

Likewise for a couple other cases below.